apps/prairielearn/src/pages/instructorGradingJob/instructorGradingJob.ts

```diff
@@ -0,0 +1,113 @@
+import ERR from 'async-stacktrace';
+import * as express from 'express';
+import { pipeline } from 'node:stream/promises';
+import { S3, NoSuchKey } from '@aws-sdk/client-s3';
+import * as error from '@prairielearn/error';
+import * as sqldb from '@prairielearn/postgres';
+import asyncHandler = require('express-async-handler');
+import type * as stream from 'node:stream';
+
+import { makeS3ClientConfig } from '../../lib/aws';
+import { InstructorGradingJob, GradingJobQueryResultSchema } from './instructorGradingJob.html';
+
+const router = express.Router();
+const sql = sqldb.loadSqlEquiv(__filename);
+
+router.get(
+  '/:job_id',
+  asyncHandler(async (req, res, next) => {
+    const gradingJobQueryResult = await sqldb.queryOptionalRow(
+      sql.select_job,
+      {
+        job_id: req.params.job_id,
+        course_instance_id: res.locals.course_instance?.id ?? null,
+        course_id: res.locals.course.id,
+        authz_data: res.locals.authz_data,
+        req_date: res.locals.req_date,
+      },
+      GradingJobQueryResultSchema,
+    );
+    if (gradingJobQueryResult === null) {
+      throw error.make(404, 'Job not found');
+    }
+
+    // If the grading job is associated with an assessment instance (through a
+    // submission, a variant, and an instance question), then we need to check
+    // if the effective user is authorized to view this assessment instance.
+    //
+    // The way we implement this check right now with authz_assessment_instance
+    // is overkill, yes, but is easy and robust (we hope).
+    if (gradingJobQueryResult.aai && !gradingJobQueryResult.aai.authorized) {
+      return next(error.make(403, 'Access denied (must be a student data viewer)'));
```

Comment:
As mentioned in a previous comment, we should just `throw` errors. There are other similar lines that I won't explicitly comment on.

Reply:
Resolved here and in 4 other lines in commit [4dc36e8](https://github.com/PrairieLearn/PrairieLearn/pull/8803/commits/4dc36e8074114af40abc3e90ba91e4077568faae).